### PR TITLE
Cleanup: better diagnostics from run_cgroups/memory

### PR DESCRIPTION
**Pull request by edsantiago**
I lost much of a day due to missing error messages from
this test. Let's avoid having that happen again.

Signed-off-by: Ed Santiago <santiago@redhat.com>

### CODE REVIEW

File: subtests/docker_cli/run_cgroups/memory.py [hunk 1]
```diff
@@ -170,6 +170,10 @@ def run_once(self):
         for subargs in self.sub_stuff['subargs']:
             dkrcmd = DockerCmd(self, 'run -d -i', subargs)
             dkrcmd.execute()
+            if self.config['expect_success'] == 'PASS':
+                mustpass(dkrcmd)
+            else:
+                mustfail(dkrcmd, 125)
```
Comment (cevich) [inline subtests/docker_cli/run_cgroups/memory.py]: Is that exit specific to 1.10?  Meh, nobody cares about 1.9 now.  nevermind.

File: subtests/docker_cli/run_cgroups/memory.py [hunk 2]
```diff
@@ -185,7 +189,11 @@ def check_result(self, subargs, long_id):
         return self.check_memory(memory_value, cgroup_memory, memory_unit)
 
     def postprocess_positive(self, this_result, passed):
-        self.failif(not isinstance(this_result, dict))
+        if this_result is None:
+            raise xceptions.DockerTestError("Invoked with null results")
+        self.failif(not isinstance(this_result, dict),
+                    ("expected this_result to be a dict; it's a %s" %
+                     this_result.__class__))
```
Comment (cevich) [inline subtests/docker_cli/run_cgroups/memory.py]: Nice!  Yeah, inconsistent typing like this is soooo confusing when reading a test.

File: subtests/docker_cli/run_cgroups/memory.py (hunk 1 above)
Comment (edsantiago) [inline subtests/docker_cli/run_cgroups/memory.py]: `mustfail()` handles 1.9 / 1.10 exit codes

File: subtests/docker_cli/run_cgroups/memory.py [hunk 3]
```diff
@@ -239,7 +249,8 @@ def postprocess(self):
                 long_id = memory_container.stdout.splitlines()[0].strip()
                 # Throws DockerIOError
                 this_result = self.check_result(subargs, long_id)
-            except (IndexError, xceptions.DockerIOError):
+            except (IndexError, xceptions.DockerIOError) as e:
+                self.logwarning("Ignoring exception: %s" % e)
```
Comment (cevich) [inline subtests/docker_cli/run_cgroups/memory.py]: good catch.

File: subtests/docker_cli/run_cgroups/memory.py (hunk 1 above)
Comment (cevich) [inline subtests/docker_cli/run_cgroups/memory.py]: awesome

**Comment by cevich** [comment]
Okay, LGTM.  Was going to merge, but I'll leave it to you in case there's anything last-minute.

**Comment by edsantiago** [comment]
Nope, nothing last-minute; I finished it last night and let it sit overnight so any bugs would settle :)

**Pull request merged**